application/classes/Ushahidi/Repository/Contact.php

```diff
@@ -126,6 +133,22 @@ public function getByContact($contact, $type)
 		return $this->getEntity($this->selectOne(compact('contact', 'type')));
 	}
 
+    public function isInTargetedSurvey($contact_id)
+    {
+        $query = DB::select('targeted_survey_state.contact_id', 'targeted_survey_state.form_id')
```

Comment:
this seems to go completely outside of the usage of contact post state to determine if a contact is indeed part of a targeted survey. We should at least consider merging both needs into a single table or it'll get very confusing.